.clang-format

```diff
@@ -58,7 +59,7 @@ CommentPragmas: '^ IWYU pragma:'
 ConstructorInitializerAllOnOneLineOrOnePerLine: false
 ConstructorInitializerIndentWidth: 4
 ContinuationIndentWidth: 4
-Cpp11BracedListStyle: false
+Cpp11BracedListStyle: true
```

Comment:
Shouldnt it be in a separate PR?

Reply:
Yes. It should. I'll do a separate PR when I'm content with automatic formatting.

Reply:
When I try it there is still a problem with long argument lists, they indent far to the left while the kernel style guide pushes them far to the right (so that the indent lines up with the first argument), e.g:
```
void __wrap_finished_web_request_statistics(uint64_t dt, uint64_t bytes_received, uint64_t bytes_sent,
                                            uint64_t content_size, uint64_t compressed_content_size) {
}
```
... gets mangled into ...
```
void __wrap_finished_web_request_statistics(
    uint64_t dt, uint64_t bytes_received, uint64_t bytes_sent, uint64_t content_size, uint64_t compressed_content_size)
{

```

Reply:
For linux kernel style (that matches the wording in the style guide and the random files that I checked):
```
AlignAfterOpenBracket: true
BinPackArguments: true
BinPackParameters: true
AllowAllParametersOfDeclarationOnNextLine: false
```

Reply:
I don't have a strong opinion on `AlignAfterOpenBracket` and `AllowAllParametersOfDeclarationOnNextLine`. But I strongly dislike `BinPackArguments: true` and `BinPackParameters: true`. Code becomes unreadable when there are multiple arguments in a row if there are multiple rows.

Reply:
Why are we changing clang-format here? Didn't we vote in the previous meet-up for the various options and agree on them?

Reply:
The reasons that it keeps coming up are:
* The docs do not match the decision made at the meet-up so the current coding style is an opaque mess known only to the people that discussed it in person.
* None of the code-base matches the coding-style so it is a problem on every commit.
* We all need to internalize the coding-style so that it is natural - this is hard when it is not documented.

It is true that it does not relate to the work on the Exporting Engine and I think that @vlvkobal is going to create a separate issue to handle this. After a long discussion on slack, I agree with @vlvkobal about the binpacking and the rest of the conversation can wait until there is an appropriate issue. I have some suggestions on how to fix the docs, but they can wait for the new issue.

Reply:
`clang-format` is the only official decision that has been made regarding code-style.
- The docs are wrong, they should be fixed.
- code-base is wrong, will be fixed by a massive code-style commit as has already been decided

Reply:
Perhaps we should continue this discussion in an RFC issue about the proposed changes to the `.clang-format` ?

Reply:
Fine by me but the changes to clang-format should be reverted.

Reply:
Yes. It's not the right place to discuss the format. Let's do it in #7271.